src/api/write.rs

```diff
@@ -84,7 +105,38 @@ impl Client {
             Ok::<_, io::Error>(buffer.split().freeze())
         });
 
-        let body = Body::wrap_stream(body);
+        #[cfg(feature = "gzip")]
+        {
+            use crate::Compression;
+            use async_compression::tokio::bufread::GzipEncoder;
+            use async_compression::Level;
+            use reqwest::header::HeaderValue;
+            use tokio_util::io::{ReaderStream, StreamReader};
+
+            match self.compression {
+                Compression::Gzip => {
+                    let encoder = GzipEncoder::with_quality(StreamReader::new(body), Level::Best);
```

Comment:
It might be better to have the compression `Level` configurable, though we're not seeing a high CPU usage and I'd figure for most people this is a decent tradeoff.